### PR TITLE
Calculating netstakeweight for given block height

### PR DESCRIPTION
**Pull request by only1question**
## Standards checklist:

- [ ] The PR title is descriptive.
- [ ] The PR doesn't replicate another PR which is already open.
- [ ] The PR has self-explained commits history.

- [ ] The code is mine or it's from somewhere with an Apache-2.0 compatible license.
- [ ] The code is efficient, to the best of my ability, and does not waste computer resources.
- [ ] The code is stable and I have tested it myself, to the best of my abilities.
- [ ] If the code introduces new classes, methods, I provide a valid use case for all of them.

## Changes:

- Added method getnetstakeweight that returns netstakeweight for required given block height

## Other comments:

...

### CODE REVIEW

File: src/pos.cpp [hunk 1]
```diff
@@ -97,6 +97,44 @@ double GetPoSKernelPS()
     return result;
 }
 
+double GetPoSKernelPS(int nHeight)
+{
+    int nPoSInterval = 72;
+    double dStakeKernelsTriedAvg = 0;
+    int nStakesHandled = 0, nStakesTime = 0;
+
+    CBlockIndex *pindex = ChainActive()[nHeight];
```
Comment (andyoknen) [inline src/pos.cpp:106]: It might break if you pass -1 here or >MaxHeight ?

File: src/rpc/net.cpp [hunk 2]
```diff
@@ -865,6 +865,44 @@ static RPCHelpMan getstakinginfo()
     };
 }
 
+static RPCHelpMan getnetstakeweight() {
```
Comment (andyoknen) [inline src/rpc/net.cpp]: I propose to correct for alignment in 4 spaces

File: src/pos.cpp (hunk 1 above)
Comment (only1question) [reply src/pos.cpp:106]: This function is called only [here](https://github.com/pocketnetteam/pocketnet.core/blob/faa0292f3f202dcce4aab1da1c298775493d16c4/src/rpc/net.cpp#L894)
And [here](https://github.com/pocketnetteam/pocketnet.core/blob/faa0292f3f202dcce4aab1da1c298775493d16c4/src/rpc/net.cpp#L889-L892)  nHeight is validated

Comment (andyoknen) [reply src/pos.cpp:106]: If we call somewhere else, then it will be necessary to do the check at the top again.

**Review by andyoknen — APPROVED** [review]

**Review by lostystyg — APPROVED** [review]

**Pull request merged**